src/js/tabs/settingsgateway.controller.js

```diff
@@ -58,8 +58,15 @@ SettingsGatewayTab.prototype.angular = function(module)
         case 'defaultRippleFlag':
           // Need to set flag on account_root
           var tx = network.remote.transaction();
-          tx.accountSet(id.account);
-          tx.setFlags('DefaultRipple');
+          var isDefaultRippleEnabled = $scope.account.Flags & ripple.Remote.flags.account_root.DefaultRipple;
+
+          if (isDefaultRippleEnabled) {
+            tx.accountSet(id.account, undefined, 'DefaultRipple');
+            $scope.options.defaultRippleFlag = false;
```

Comment:
$scope.options refers to Options in config.js. What does user account flag have to do with client configuration?